deepspeed/runtime/zero/stage3.py

```diff
@@ -1457,8 +1457,8 @@ def complete_grad_norm_calculation_for_cpu_offload(self, params):
         norm_is_nan = total_norm.isnan()
         inf_or_nan = norm_is_nan.logical_or(norm_is_inf)
 
-        err = torch.tensor(-1.0, device=inf_or_nan.device, dtype=torch.float)
-        total_norm = inf_or_nan * err + inf_or_nan.logical_not() * total_norm
+        err = torch.tensor(-1.0, device=inf_or_nan.device, dtype=torch.float)        
```

Comment:
@nelyahu, given the repetition of this sequence, does it make sense to refactor into a utility function?

Reply:
I am reffering to the following specifically: 

```python
        total_norm = total_norm_cuda**(1. / norm_type)
        norm_is_inf = total_norm.isinf()
        norm_is_nan = total_norm.isnan()
        inf_or_nan = norm_is_nan.logical_or(norm_is_inf)
        err = torch.tensor(-1.0, device=self.device, dtype=torch.float)        
        total_norm = torch.where(inf_or_nan, err, total_norm)
```